src/dishka/container.py

```diff
@@ -156,6 +157,8 @@ def get(
             component: Component | None = DEFAULT_COMPONENT,
     ) -> Any:
         lock = self.lock
+        if is_type_alias_type(dependency_type):
+            dependency_type = dependency_type.__value__
```

Comment:
Can it be that `__value__` is also a DependencyAlias? Do we need to unpack it too?